content/en/logs/guide/collect-google-cloud-logs-with-push.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Collect Google Cloud Logs with a Pub/Sub Push Subscription
+kind: documentation
+further_reading:
+- link: "/logs/explorer/"
+  tag: "Documentation"
+  text: "Learn how to explore your logs"
+- link: "/logs/explorer/#visualize"
+  tag: "Documentation"
+  text: "Perform Log Analytics"
+- link: "/logs/log_configuration/processors"
+  tag: "Documentation"
+  text: "Learn how to process your logs"
+---
+
+<div class="alert alert-danger">
+This page describes deprecated features with configuration information relevant to legacy Pub/Sub Push subscriptions, useful for troubleshooting or modifying legacy setups. Use a <strong>Pull</strong> subscription with the Datadog Dataflow template to forward your Google Cloud logs to Datadog instead. See <a href="https://docs.datadoghq.com/integrations/google_cloud_platform/#log-collection" target="_blank">Log collection</a> on the Google Cloud integration page for instructions.
+</div>
+
+## Overview
+
+This guide describes how to forward logs from your Google Cloud services to Datadog through a **Push** subscription to a [Google Cloud Pub/Sub][10] topic.
+
+To collect logs from applications running in GCE or GKE, you can also use the [Datadog Agent][13].
+
+**Note**: If you have a [Google Cloud VPC][14] in your Google Cloud environment, the **Push** subscription cannot access endpoints outside the VPC.
+
+## Setup
+
+### Prerequisites
+
+The [Google Cloud Platform integration][1] is successfully installed.
+
+### Create a Cloud Pub/Sub topic
+
+1. Go to the [Cloud Pub/Sub console][2] and create a new topic.
+
+    {{< img src="integrations/google_cloud_platform/create_a_topic.png" alt="Create a topic" style="width:80%;">}}
```

Comment:
Interesting, when I go to the Google Cloud console to create a topic, I don't get this image? I instead get what is shown [here](https://docs.datadoghq.com/integrations/google_cloud_platform/#create-a-cloud-pub-sub). Is this an outdated image?

Another note: in the page I linked above, it is blocking most of the configuration with the help message for "Add a default subscription". Customers **should not** add a default subscription, bc the default is a pull subscription, and this solution requires a push subscription.

But FYI the "Add a default subscription" is checked by default. Here's what I see:
<img width="559" alt="image" src="https://github.com/DataDog/documentation/assets/127151740/0ba88939-c6a9-4aa2-8b24-1854735441b5">

Reply:
To be clear, for the Dataflow set up, customers **should** use the default subscription, since Dataflow uses a pull subscription.

For this outdated set up, customers **should not** use the default subscription, since this set up works via a push subscription sending logs via HTTPS directly to Datadog.

Reply:
Thanks for flagging that and providing the additional context. It is indeed an outdated screenshot, and updated in [025413c](https://github.com/DataDog/documentation/pull/19467/commits/025413c3d479d02827aad844dafbd8ff8c0037bc)